module/icp/algs/modes/modes.c

```diff
@@ -154,3 +154,43 @@ crypto_free_mode_ctx(void *ctx)
 		kmem_free(ctx, sizeof (gcm_ctx_t));
 	}
 }
+
+static void *
+explicit_memset(void *s, int c, size_t n)
```

Comment:
See also the discussion in https://github.com/systemd/systemd/pull/25973. It'd probably be worth looking at using the C23 `explicit_memset` if available and falling back to e.g. `explicit_bzero` if it exists.

This is what shadow does: https://github.com/shadow-maint/shadow/blob/master/lib/defines.h#L50.

Reply:
We are currently using C99 and were still using C89 several years ago. I estimate that we will not adopt C23 until 2040 at the earliest.

That said, C23 introduces `memset_explicit()` in the standard libraries. Being a kernel module means that we will not have access to standard libraries and the way that our cross platform support is implemented basically means that we are emulating OpenSolaris, which will never get `memset_explicit()`, although maybe illumos will. We can always revisit this in the future after we see what the various platforms do.